### PR TITLE
Include all matching component schematics in Ingredient Selector

### LINKED ISSUE #58: `getComponentList.py` excludes manually-entered components
Manually-entered component schematics are not included in this list, and therefore not available as components in other schematics. For example:

1. Create a new schematic for an Armor Component, choosing manual entry
    + set **Crafting Tab** to `Armor`
    + set **Object Type** to `Armor Schematic`
2. Create a new armor schematic
3. Edit the armor schematic
4. Choose "Component" from the Ingredient Selector
5. Choose "Armor" from the filter

You won't see the Armor Component from step 1.

This also applies to any components copied from manually-entered component schematics.

---

Diving in a bit, it looks like there's two conditions for a component schematic to be included in the list:

- its `tSchematic.craftingTab` matches the filtered crafting tab (required)
- either:
    + `tSchematic.objectGroup` contains `"component"` in the path, or
    + `tSchematic.objectType` is equal to `262144` (one of the generic "Component" types)

The problem here is that manually-entered schematics do not set an `objectGroup`. The code sets it to an empty string. Therefore it can never match `%component%`. A possible workaround is to select the "_correct_" `objectType` (there are 4 or 5 of them).

As for solutions, I think we could:

1. join on `tObjectType` and add an `OR` condition where `tObjectType.typeName LIKE "%component%"`
2. extend the condition for `tSchematic.objectType` to use an `IN` syntax with a list of object type IDs

Personally, I'm inclined to start with option 1, as its fairly obvious (e.g. any object type with the word "component" in it) and gives us backward-and-forward-compatibility. I'm going to open a PR with this proposed change, but open to other proposals!

### PR DESCRIPTION
**Pull request by cbortz**
Fixes #58

### CODE REVIEW

**Comment by cbortz** [comment]
@pwillworth please take a look when you can 🙏

File: html/getComponentList.py [hunk 1]
```diff
@@ -36,14 +36,27 @@
...
 
 conn = dbShared.ghConn()
 cursor = conn.cursor()
 if (cursor):
-	cursor.execute('SELECT tSchematic.schematicID, schematicName, objectPath, imageName FROM tSchematic LEFT JOIN (SELECT schematicID, imageName FROM tSchematicImages WHERE imageType=1) tsi ON tSchematic.schematicID=tsi.schematicID WHERE (objectPath IN (SELECT ingredientObject FROM tSchematicIngredients WHERE ingredientType>0) OR objectGroup LIKE "%component%" OR objectType=262144)' + criteriaStr + ' ORDER BY schematicName;')
+	queryStr = f"""
+	SELECT tSchematic.schematicID, schematicName, objectPath, imageName
+	FROM tSchematic
+	LEFT JOIN (SELECT schematicID, imageName FROM tSchematicImages WHERE imageType = 1) tsi ON tSchematic.schematicID = tsi.schematicID
+	LEFT JOIN tObjectType ON tObjectType.objectType = tSchematic.objectType
```
Comment (cbortz) [inline html/getComponentList.py:50]: This join is new.

File: html/getComponentList.py [hunk 2]
```diff
@@ -36,14 +36,27 @@
...
+	WHERE (
+		objectPath IN(SELECT ingredientObject FROM tSchematicIngredients WHERE ingredientType > 0)
+		OR objectGroup LIKE "%component%"
+		OR tObjectType.typeName LIKE "%component%"
```
Comment (cbortz) [inline html/getComponentList.py:54]: This condition is new.

**Review by cbortz — COMMENTED** [review]
Highlighting critical additions. Everything else is formatting & syntax (e.g. prefixing now-ambiguous columns with a table name)

**Comment by cbortz** [comment]
Build is failing currently, but would be fixed via rebase over #61 if/when merged

**Review by pwillworth — APPROVED** [review]
This seems like a good solution to me.  Adds the functionality from raw schematic creation without having to add a selection for objectGroup which is largely a behind the scenes thing.  Have deployed this change to test.galaxyharvester.net and verified it.

**Comment by cbortz** [comment]
@pwillworth Build is now passing. I saw your approval. I'll leave you to do the merging, etc. 🙏 Thank you for accepting this.

**Pull request merged**